File: VaccineDrugLocation/vaccine_drug_location_backend/vaccine_drug_location/src/main/java/dke/vaccine_location_drug/entity/Location.java
package dke.vaccine_location_drug.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import java.util.List;

@Entity
public class Location {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long location_id;
    private String name;
    private String county;
    private String address;
    private String type;
    private int duration;

    @OneToMany(mappedBy = "location", cascade = CascadeType.ALL)
    @JsonIgnore
    private List<Line> lines;

    // Getter und Setter

    public Long getId() {
        return location_id;
    }

    public void setId(Long id) {
        this.location_id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getCounty() {
        return county;
    }

    public void setCounty(String county) {
        this.county = county;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public List<Line> getLines() {
        return lines;
    }

    public int getDuration() {
        return duration;
    }

    public void setDuration(int duration) {
        this.duration = duration;
    }

}

File: vaccine_drug_location_frontend/vaccine_drug_location/src/app/entities/Location.ts
import {Line} from "./Line";

export class Location {
  id: number;
  name: string;
  county: string;
  address: string;
  type: string;
  duration: number;
  lines: Line[];

  constructor(
    id: number,
    name: string,
    county: string,
    address: string,
    type: string,
    duration: number,
    lines: Line[] = []
  ) {
    this.id = id;
    this.name = name;
    this.county = county;
    this.address = address;
    this.type = type;
    this.duration = duration;
    this.lines = lines;
  }
}
